pointcloud/basic_octree.go

```diff
@@ -132,9 +134,86 @@ func (octree *BasicOctree) MetaData() MetaData {
 	return octree.meta
 }
 
-// CollidesWithGeometry will return whether a given geometry is in collision with a given point.
-func (octree *BasicOctree) CollidesWithGeometry(geom spatialmath.Geometry, threshold int, buffer float64) (bool, error) {
-	if octree.MaxVal() < threshold {
+// Transform recursively steps through the octree and transforms it by the given pose.
+func (octree *BasicOctree) Transform(p spatialmath.Pose) *BasicOctree {
+	if spatialmath.PoseAlmostEqual(p, spatialmath.NewZeroPose()) {
+		return octree
+	}
+
+	var transformedOctree *BasicOctree
+	switch octree.node.nodeType {
+	case internalNode:
+		newCenter := octree.center.Add(p.Point())
```

Comment:
This isn't correct to do and will lead to the octree being in the wrong location. `center` must be a pose for this.

I would pause work here until we've decided on the path forwards for octrees

Reply:
Yeah we would need this be a `Pose`.  I think it would also be fine to remove this from scope for the time being and just leave as a stub, since it isn't being called at the moment anyway

Reply:
This is actually called as part of TP-space.

Reply:
Also, can we confirm whether generated pointclouds are given in the frame of the observer (3d camera or lidar) or in the frame of world?

If the former, this will need to be called at least once to transform an octree into world frame for querying.

Reply:
going to be leaving this as a stub to be implemented in a later ticket